src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -5345,23 +5345,23 @@ trait Types
   @tailrec
   private[scala] final def typeIsNothing(tp: Type): Boolean =
     tp.dealias match {
-      case PolyType(_, tp) => typeIsNothing(tp)
+      case PolyType(_, resultType) => typeIsNothing(resultType)
```

Comment:
these warnings go away by taking tp in the selection `tp.dealias` (and the types are OK). Very nice.